tools/compile_tf_graph.py

```diff
@@ -461,13 +460,25 @@ def add_stochastic_var(self, name):
 
   def _get_cell(self, unit, unit_opts=None):
     """
-    :param str|dict[str] unit:
+    Replace the unit by an identical SubnetworkRecCellSingleStep instance
+    Only allows for SubnetworkRecCell, not strings as the super method does
```

Comment:
We should not make a new instance/copy.
I'm pretty sure there is several code which expects that it stays the same instance.
I'm also pretty sure such copy could potentially introduce some very nasty edge cases and buggy behavior.

Rather, this `RecStepByStepLayer` should directly create a `SubnetworkRecCellSingleStep` instance in the first place, instead of a `_SubnetworkRecCell`.
This creation happens in `RecLayer.transform_config_dict` via `subnet = _SubnetworkRecCell(...)`.
I just pushed 416209a2d8929667b7e8011d0413d0706d33ae0a which allows you to easily make use of `SubnetworkRecCellSingleStep` in the same way.

Reply:
I think this whole function `_get_cell` can be removed here then.

Reply:
(So you need to rebase the branch / PR on current master.)

Reply:
I did not want to mess with this without your comments on that, so I will check now how that works with your change.